src/eredis_cluster_manager.erl

```diff
@@ -0,0 +1,415 @@
+-module(eredis_cluster_manager).
+-behaviour(gen_server).
+
+%% API.
+-export([start_link/1]).
+-export([remap_cluster/2]).
+-export([get_eredis_pid/2]).
+-export([qp/2]).
+-export([q/2]).
+
+%% gen_server.
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([terminate/2]).
+-export([code_change/3]).
+
+%% Type definition.
+-include("eredis_cluster.hrl").
+-record(state, {
+    init_nodes :: [#node{}],
+    slots :: tuple(), %% Tuple with all the Redis slots. Each element is index into slots_maps.
+    slots_maps :: tuple(), %% Tuple with all the Redis nodes and Redis slots mapped on them.
+    version :: integer()  %% Used to avoid unnecessary refresh of Redis slots.
+}).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% API
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec start_link([]) -> {ok, pid()}.
+start_link({ModName, Args}) ->
+    gen_server:start_link({local, ModName}, ?MODULE, [Args], []).
+
+%% Looks up eredis Pid for Slot.
+-spec get_eredis_pid(Cluster::any(), Slot::integer()) -> 
+    {Pid::pid() | undefined, Version::integer()}.
+get_eredis_pid(Cluster, Slot) ->
+    {ok, Result} = gen_server:call(Cluster, {get_pid_by_slot, Slot}),
+    Result.
```

Comment:
usually the name of the function is the same as the message send to the gen_server. Also the result is returned as is. aka if you don't want the ok don't return it from the gen_server